md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessLogDAO.java

```diff
@@ -282,5 +306,32 @@ public ProcessLogInfo getLastValue(ProcessLogInfo processLogInfo) {
         return processLogInfoList;
     }
 
+    public List<ProcessLogInfo> listLastInstanceRef(int processId, String messageId) {
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        Criteria criteriaList = session.createCriteria(ProcessLog.class).add(Restrictions.eq("process.processId",processId)).add(Restrictions.eq("messageId",messageId));
+        Criteria criteriaMaxInstanceRef = session.createCriteria(ProcessLog.class).add(Restrictions.eq("process.processId",processId)).add(Restrictions.eq("messageId",messageId)).addOrder(Order.desc("instanceRef")).setMaxResults(1);
+        ProcessLog maxInstanceRef=criteriaMaxInstanceRef.list().size()>0?(ProcessLog)criteriaMaxInstanceRef.list().get(0):null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1155)